account_payment_other_company/models/account_payment.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (C) 2019 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountPayment(models.Model):
+    _inherit = 'account.payment'
+
+    other_journal_id = fields.Many2one('account.journal',
+                                       string='Paid By')
+    hide = fields.Boolean(default=True,
+                          invisible=True)
+
+    @api.multi
+    def action_validate_invoice_payment(self):
+        res = super().action_validate_invoice_payment()
+        if not self.hide:
+            account_move = self.env['account.move']
+            vals = {
+                'journal_id': self.other_journal_id.id,
+                'state': 'draft',
+                'company_id': self.other_journal_id.company_id.id,
+                'ref': self.id,
+                # move.partner_id is empty with this line of code?
+                # 'partner_id': self.partner_id.id,
+                'line_ids': [(0, 0, {
+                    'account_id': self.
+                    other_journal_id.company_id.due_from_account_id.id,
+                    'partner_id': self.company_id.partner_id.id,
+                    'credit': self.amount,
+                }), (0, 0, {
+                    'account_id': self.
+                    other_journal_id.default_debit_account_id.id,
+                    'partner_id': self.partner_id.id,
+                    'debit': self.amount,
+                })]
+            }
```

Comment:
@osi-scampbell Add a new method to prepare the values and call it from here. It will make the code easier to inherit.